src/ast.cpp

```diff
@@ -87,7 +87,39 @@ void SelectCommand::execute(Database& database)
 
 void DeleteFromCommand::execute(Database& database)
 {
-  std::cout << *this << std::endl;
+  using namespace white::davisbase::sdl;
+  auto table_opt = database.getTable(table_name);
+  if (!table_opt.has_value()) {
+    throw std::runtime_error("Table doesn't exist");
+  }
+
+  if (condition == std::nullopt) {
+    throw std::runtime_error("Where clause required");
+  }
+
+  auto& table = table_opt.value();
+  auto column_defs = table.columnDefinitions();
+  std::map<std::string, common::ColumnDefinition> columns_map;
+
+  for (int i = 0; i < column_defs.size(); i++) {
+    columns_map[column_defs[i].name] = column_defs[i];
+  }
+
+  if (columns_map.find(condition->column_name) != columns_map.end()) {
+    throw std::runtime_error("Column doesn't exist");
+  }
+
+  auto colValVariant = createColumnValue(
+    columns_map[condition->column_name].type, condition->literal);
+
+  table.mapOverRecords(
+    [&](CellIndex i, TableLeafPage& page, TableLeafCell cell) {
+      if (isWhereSatisfied(colValVariant, condition.value())) {
```

Comment:
Important Detail: In this step, we would like to check the column value against condition. Let's say `condition->column_name` is indice `col_idx` of `column_defs`. `col_idx` can be computed before calling lambda, since it's always same. Then, we need to check `isWhereSatisfied` against `cell.row_data[col_idx]` and `condition.value()`.

Tiny Detail: `condition.value()` will always check if it is `nullopt` in the underlying implementation and throw exception if it is so. Since we are sure it isn't so at this point, you can store it in a local reference before `mapOverRecords` to avoid this overhead everytime so that we can provide best performance to customers of white-davisbase :P

Reply:
Holy moly.

I just [learned](https://en.cppreference.com/w/cpp/utility/optional/operator*) if you use `*condition` there is no overhead at all! Neat.